contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -304,6 +338,26 @@ fn route_to_nexus(
         .into_iter()
         .chain(iter::once(client.route_message(msg)))
         .collect::<Vec<_>>()
-        .then(|msgs| Response::new().add_messages(msgs))
+        .then(Ok)
+}
+
+pub fn route_messages_to_nexus(
+    client: &nexus::Client,
+    nexus: &Addr,
+    msgs: Vec<Message>,
+) -> Result<Response<nexus::execute::Message>> {
+    let nexus_msgs = msgs
+        .clone()
+        .into_iter()
+        .map(|msg| route_to_nexus(client, nexus, msg, None))
```

Comment:
Consider merge this function with `route_to_nexus` since `token` shouldn't be relevant anymore?

Reply:
sure, plan to drop token in a separate pr, can merge there